src/app/hooks/useScreensize.ts

```diff
@@ -1,12 +1,40 @@
-import useMediaQuery from '@mui/material/useMediaQuery'
-import { useTheme } from '@mui/material/styles'
+import { useState, useLayoutEffect } from 'react'
+
+// Sync with Shadcn/Oasis UI Library breakpoints
+const BREAKPOINTS = {
+  md: 768,
+  lg: 1024,
+  xl: 1280,
+}
 
 export const useScreenSize = () => {
-  const theme = useTheme()
-  return {
-    isMobile: useMediaQuery(theme.breakpoints.down('md')),
-    isTablet: useMediaQuery(theme.breakpoints.down('lg')),
-    isLaptop: useMediaQuery(theme.breakpoints.down('xl')),
-    isDesktop: useMediaQuery(theme.breakpoints.up('xl')),
-  }
+  const [screenSize, setScreenSize] = useState({
+    isMobile: false,
+    isTablet: false,
+    isLaptop: false,
+    isDesktop: false,
+  })
+
+  useLayoutEffect(() => {
+    const handleResize = () => {
+      const isMobile = window.matchMedia(`(max-width: ${BREAKPOINTS.md - 1}px)`).matches
+      const isTablet = window.matchMedia(`(max-width: ${BREAKPOINTS.lg - 1}px)`).matches
+      const isLaptop = window.matchMedia(`(max-width: ${BREAKPOINTS.xl - 1}px)`).matches
+      const isDesktop = window.matchMedia(`(min-width: ${BREAKPOINTS.xl}px)`).matches
+
+      setScreenSize({
+        isMobile,
+        isTablet,
+        isLaptop,
+        isDesktop,
+      })
+    }
+
+    window.addEventListener('resize', handleResize)
```

Comment:
Previous mui implementation used `mediaQueryList.addEventListener('change', notify);`
https://github.com/mui/material-ui/blob/dba602459c8b50d34ecbf3c6e61a3f805c7cb413/packages/mui-system/src/useMediaQuery/useMediaQuery.ts#L103-L108

to consider if we see any issues

Reply:
switched to media query change listener